config/auth_azure_github_oidc_test.go

```diff
@@ -0,0 +1,287 @@
+					},
+					"POST /test-tenant-id/oauth2/token": {
+						Status: http.StatusOK,
+						ExpectedHeaders: map[string]string{
+							"Accept":       "application/json",
+							"Content-Type": "application/x-www-form-urlencoded",
+						},
+						Response: map[string]string{
+							"foo": "bar",
+						},
+					},
+				},
+			},
+			wantErrPefix: errPrefix("inner token: invalid token"),
+		},
+		{
+			desc: "success",
+			cfg: &Config{
+				DatabricksEnvironment: &environment.DatabricksEnvironment{
+					Cloud:              environment.CloudAzure,
+					AzureApplicationID: "test-azure-app-id",
+					AzureEnvironment:   &environment.AzurePublicCloud,
+				},
+				AzureClientID:              "test-client-id",
+				AzureTenantID:              "test-tenant-id",
+				ActionsIDTokenRequestURL:   "http://endpoint.com/test?version=1",
+				ActionsIDTokenRequestToken: "token-1337",
+				Host:                       "http://host.com/test",
+				HTTPTransport: fixtures.MappingTransport{
+					"GET /test?version=1&audience=api://AzureADTokenExchange": {
+						Status: http.StatusOK,
+						ExpectedHeaders: map[string]string{
+							"Authorization": "Bearer token-1337",
+							"Accept":        "application/json",
+						},
+						Response: `{"value": "id-token-42"}`,
+					},
+					"POST /test-tenant-id/oauth2/token": {
+						Status: http.StatusOK,
+						ExpectedHeaders: map[string]string{
+							"Accept":       "application/json",
+							"Content-Type": "application/x-www-form-urlencoded",
+						},
+						Response: map[string]string{
+							"token_type":    "access-token",
+							"access_token":  "test-auth-token",
+							"refresh_token": "refresh",
+							"expires_on":    "0",
+						},
+					},
+				},
+			},
+			wantHeaders: map[string]string{
+				"Authorization": "access-token test-auth-token",
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.desc, func(t *testing.T) {
+			tc.cfg.Credentials = &AzureGithubOIDCCredentials{} // only test this credential strategy
+			tc.cfg.DebugHeaders = true
+			if tc.wantHeaders == nil {
+				tc.wantHeaders = map[string]string{}
+			}
+
+			req, _ := http.NewRequest("GET", "http://localhost", nil)
+			gotErr := tc.cfg.Authenticate(req)
+			gotHeaders := map[string]string{}
+			for h := range req.Header {
+				gotHeaders[h] = req.Header.Get(h)
+			}
+
+			if tc.wantErrPefix == nil && gotErr != nil {
+				t.Errorf("Authenticate(): got error %q, want none", gotErr)
+			}
+			if tc.wantErrPefix != nil && !hasPrefix(gotErr, *tc.wantErrPefix) {
+				t.Errorf("Authenticate(): got error %q, want error with prefix %q", gotErr, *tc.wantErrPefix)
+			}
+			if diff := cmp.Diff(tc.wantHeaders, gotHeaders); diff != "" {
+				t.Errorf("Authenticate(): mismatch (-want +got):\n%s", diff)
+			}
+		})
+	}
+}
+
+func TestAzureGithubOIDCCredentials_Name(t *testing.T) {
+	c := AzureGithubOIDCCredentials{}
+	want := "github-oidc-azure"
+
+	if got := c.Name(); got != want {
```

Comment:
Just curious, why not `assert.Equal(t, "github-oidc-azure", c.Name())`?

Reply:
It's a good question. The short answer is that using `testify` is not idiomatic. As a result, one could argue that using `testify` actually goes against Databricks' Go style guidelines (see refs below). 

IMO, `testify` can be quite convenient in some use cases but it adds cognitive load for the developers and the readers that are not familiar with the library (i.e. it is one additional mini-language to learn). Adding that cognitive load doesn't seem to be worth the couple of lines it removes.

Some references:

- https://google.github.io/styleguide/go/decisions#assertion-libraries (not in Databricks' go style)
- https://go.dev/wiki/CodeReviewComments#useful-test-failures